ops/roles/cgus/tasks/database.yml

```diff
@@ -0,0 +1,44 @@
+- name: Remove index.lock in {{ name }}
+  file:
+    path: "/home/{{ ansible_user }}/{{ directory }}/.git/index.lock"
+    state: absent
+
+- name: Create backup tag for {{ name }}
+  command:
+    cmd: git tag {{ ansible_date_time.iso8601_basic_short }}
+    chdir: "/home/{{ ansible_user }}/{{ directory }}"
+  tags:
+    - restart
+    - start
+    - update
+
+- name: Clone or update {{ name }} repository
+  git:
+    repo: "{{ repository }}"
+    dest: "/home/{{ ansible_user }}/{{ directory }}"
+    depth: 1
+    force: yes
+    accept_hostkey: yes
+    key_file: "/home/{{ ansible_user }}/.ssh/cgus-bot-key"
+  register: git_clone_app_finished
+  tags:
+    - setup
+    - update
+
+- name: Reset {{ name }} on origin {{ branch }}
+  command:
+    cmd: git reset --hard origin/{{ branch }}
+    chdir: "/home/{{ ansible_user }}/{{ directory }}"
```

Comment:
If we want to always blindly trust the remote, then the following commands need to be issued:

```sh
# One does not simply `git pull`: we need to clean all leftover files and support force pushes
git fetch origin $TARGET_BRANCH  # get the remote code, do not try to apply it just yet
git checkout --force --detach origin/$TARGET_BRANCH  # ignore divergences, always trust remote
git clean --force  # remove all leftover files in tracked folders, prevent conflicts with previously ignored files
```

[Source](https://gist.github.com/MattiSG/3a5e106449a0786e4454d11338633bcd#file-deploy-sh-L27-L29).